### PR TITLE
Text input validation

### PR DESCRIPTION
**Pull request by graham-dds**
Addresses: https://www.pivotaltracker.com/story/show/170601644

Adds a little more text validation for a few `StringFields()` in our forms.

It also:
 - alters the `IsNumber()` validator to allow falsey values. We should use WTForm's built-in `Requried()` validator if we want input
- Makes CLIN numbers a max of 4 digits, and alters some test data to reflect this fact

TODO in another PR: Make sure all of our text input fields have a character limit in the models

### CODE REVIEW

File: atst/app.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def make_config(direct_config=None):
         config.read_dict({"default": direct_config})
 
     # Assemble DATABASE_URI value
-    database_uri = "postgres://{}:{}@{}:{}/{}".format(  # pragma: allowlist secret
+    database_uri = "postgresql://{}:{}@{}:{}/{}".format(  # pragma: allowlist secret
```
Comment (graham-dds) [inline atst/app.py]: Screen Shot 2020-01-16 at 9 55 25 AM

File: atst/forms/application.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ class EnvironmentsForm(BaseForm):
         StringField(
             label=translate("forms.application.environment_names_label"),
             filters=[remove_empty_string],
+            validators=[AlphaNumeric(), Length(max=100)],
```
Comment (dandds) [inline atst/forms/application.py]: This is good, but I'm reluctant to enforce backend validations that we don't also have on the frontend. That's a surprise for the user when they submit the form. (Or at least I don't see where we're enforcing this on the frontend.)

File: atst/forms/portfolio.py [hunk 3]
```diff
@@ -20,10 +21,11 @@ class PortfolioForm(BaseForm):
                 min=4,
                 max=100,
                 message=translate("forms.portfolio.name.length_validation_message"),
-            )
+            ),
+            Name(),
         ],
     )
-    description = TextAreaField(translate("forms.portfolio.description.label"),)
+    description = TextAreaField(translate("forms.portfolio.description.label"))
```
Comment (dandds) [inline atst/forms/portfolio.py]: Do we want a length validator? (I don't know if the intent of this PR is to cover every string and hidden field.)

File: tests/factories.py [hunk 4]
```diff
@@ -327,7 +331,7 @@ class Meta:
         model = CLIN
 
     task_order = factory.SubFactory(TaskOrderFactory)
-    number = factory.LazyFunction(random_task_order_number)
+    number = factory.LazyFunction(random_clin_number)
```
Comment (dandds) [inline tests/factories.py]: I was wondering why I was seeing 13-digit numbers in my test database.

File: atst/forms/task_order.py [hunk 5]
```diff
@@ -129,7 +137,7 @@ class AttachmentForm(BaseForm):
             Length(
                 max=40, message=translate("forms.attachment.object_name.length_error")
             ),
-            Regexp(regex=REGEX_ALPHA_NUMERIC),
+            AlphaNumeric(),
         ],
     )
```
Comment (dandds) [inline atst/forms/task_order.py]: There are also `HiddenField` instance in `atst.forms.application_member.EnvironmentForm`; can you add this validator there, too?

**Comment by graham-dds** [comment]
@dandds I'm going to reopen this PR to add:
- frontend validation where necessary
- remaining `Length()` validations to form fields
- character limits on our models, which will add a migration

I was initially thinking of doing this in two PRs, but one probably makes sense for this

**Comment by graham-dds** [comment]
@dandds I'm going to reopen this PR to add:
- frontend validation where necessary
- remaining `Length()` validations to form fields
- character limits on our models, which will add a migration

I was initially thinking of doing this in two PRs, but one probably makes sense for this

**Pull request closed without merging**